packages/synapse-react-client/src/components/ChangePassword/ChangePassword.integration.test.tsx

```diff
@@ -0,0 +1,440 @@
+import React from 'react'
+import ChangePassword from './ChangePassword'
+import { render, screen, waitFor, within } from '@testing-library/react'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import userEvent from '@testing-library/user-event'
+import { server } from '../../mocks/msw/server'
+import { noop } from 'lodash-es'
+import * as ToastMessage from '../ToastMessage/ToastMessage'
+import { MemoryRouter } from 'react-router-dom'
+import SynapseClient from '../../synapse-client'
+import {
+  MOCK_USER_ID,
+  mockUserProfileData,
+} from '../../mocks/user/mock_user_profile'
+import {
+  getBadRequestChangePasswordHandler,
+  getRequires2FAChangePasswordHandler,
+  getSuccessfulChangePasswordHandler,
+} from '../../mocks/msw/handlers/changePasswordHandlers'
+import { BackendDestinationEnum, getEndpoint } from '../../utils/functions'
+
+const mockDisplayToast = jest
+  .spyOn(ToastMessage, 'displayToast')
+  .mockImplementation(() => noop)
+
+const changePasswordSpy = jest.spyOn(SynapseClient, 'changePassword')
+
+function renderComponent() {
+  return render(
+    <MemoryRouter>
+      <ChangePassword />
+    </MemoryRouter>,
+    { wrapper: createWrapper() },
+  )
+}
+
+function getCurrentPasswordField() {
+  return screen.getByLabelText(/Current password/i)
+}
+
+function getNewPasswordField() {
+  return screen.getByLabelText(/New password/i)
+}
+
+function getConfirmPasswordField() {
+  return screen.getByLabelText(/Confirm password/i)
+}
+
+function setUp() {
+  const user = userEvent.setup()
+  const result = renderComponent()
+  const currentPasswordField = getCurrentPasswordField()
+  const newPasswordField = getNewPasswordField()
+  const confirmPasswordField = getConfirmPasswordField()
+  const submitButton = screen.getByRole('button', { name: 'Change Password' })
+
+  return {
+    user,
+    result,
+    currentPasswordField,
+    newPasswordField,
+    confirmPasswordField,
+    submitButton,
+  }
+}
+
+describe('ChangePassword tests', () => {
+  beforeAll(() => server.listen())
+  beforeEach(() => jest.clearAllMocks())
+  afterEach(() => server.resetHandlers())
+  afterAll(() => server.close())
+
+  it('allows changing password', async () => {
+    server.use(
+      getSuccessfulChangePasswordHandler(
+        getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
+      ),
+    )
+
+    const currentPassword = 'currentPassword'
+    const newPassword = 'newPassword'
+
+    const {
+      user,
+      currentPasswordField,
+      newPasswordField,
+      confirmPasswordField,
+      submitButton,
+    } = setUp()
+
+    await user.type(currentPasswordField, currentPassword)
+    await user.type(newPasswordField, newPassword)
+    await user.type(confirmPasswordField, newPassword)
+
+    await userEvent.click(submitButton)
```

Comment:
Does `userEvent` need to be used here or could `user` be used instead? If `user` works, then could update the other instances in this test suite as well

```suggestion
    await user.click(submitButton)
```